mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -611,6 +628,46 @@ message GenerateBurnRedemptionTxResponse {
     TxProposal tx_proposal = 1;
 }
 
+// Generate a simple swap proposal. The result is a signed contingent input
+// which trades one currency for another and is suitable for use with the deqs.
+// (This API is restrictive and doesn't let you build more complex SCIs.)
+message GenerateSwapRequest {
+    // Monitor id sending the funds.
+    bytes sender_monitor_id = 1;
+
+    // Subaddress to return change to.
+    uint64 change_subaddress = 2;
+
+    // A specific input, whose value will be offered in full by the swap.
+    //
+    // You may need to conduct a self-spend first to find an input of exactly
+    // the correct value before using this API if none of your inputs match
+    // the offer you want to make.
+    UnspentTxOut input = 3;
+
+    // The u64 value we are asking for in exchange for our input
+    uint64 ask_value = 4;
+
+    // The token_id we are asking for in exchange for our input
+    uint64 ask_token_id = 5;
+
+    // If set to true, the offer is "all or nothing", the entire ask must be supplied.
+    // Otherwise, it is a "partial-fill" SCI
+    bool all_or_nothing = 6;
```

Comment:
I would change this to allow_partial_fill, since thats the term we use more often than "all or nothing"

Reply:
yeah i need to get more consistent about naming things, sorry